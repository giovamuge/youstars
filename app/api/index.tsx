// export * from "./github/stars/get/index"
// export * from "./github/token/post/index"
// export * from "./github/profile/get/index"
