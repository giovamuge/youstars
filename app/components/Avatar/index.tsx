import React from "react"
import { Image, StyleSheet } from "react-native"
import { colors } from "../../theme"

type Props = {
	uri: string
	size: number
}

export function Avatar({ uri, size = 50 }: Props): JSX.Element {
	return (
		<Image
			source={{ uri }}
			style={[
				style.image,
				{ width: size, height: size, borderRadius: size / 2 },
			]}
		/>
	)
}

const style = StyleSheet.create({
	image: {
		borderColor: colors.palette.black,
		borderWidth: 3,
		justifyContent: "center",
		alignItems: "center",
		objectFit: "cover",
	},
})
