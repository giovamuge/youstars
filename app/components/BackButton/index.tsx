import {
	NavigationProp,
	useNavigation,
} from "@react-navigation/native"
import React, { PropsWithChildren } from "react"
import { Pressable, Text } from "react-native"

type Props = PropsWithChildren<{}>

export function BackButton({ children }: Props): JSX.Element {
	const navigation = useNavigation<NavigationProp<any, any>>()
	return (
		<Pressable onPress={() => navigation.goBack()}>
			{children ? children : <Text>Back</Text>}
		</Pressable>
	)
}
