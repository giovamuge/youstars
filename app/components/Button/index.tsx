import React, { PropsWithChildren, useRef } from "react"
import { Animated, Easing, Pressable, Text, View } from "react-native"
import styles from "./style"

type Props = PropsWithChildren<{
	title: string
	disabled?: boolean
	onPress: () => void
	passedClass?: object
}>

export function Button({
	title,
	disabled,
	onPress,
	passedClass,
}: Props): JSX.Element {
	const marginAnimated = useRef(new Animated.Value(7)).current

	function animateInMarginBottom() {
		Animated.timing(marginAnimated, {
			toValue: 0,
			duration: 300,
			useNativeDriver: false,
			easing: Easing.cubic,
		}).start()
	}

	function animateOutMarginBottom() {
		Animated.timing(marginAnimated, {
			toValue: 7,
			duration: 300,
			useNativeDriver: false,
			easing: Easing.cubic,
		}).start()
	}

	return (
		<Pressable
			disabled={disabled}
			onPress={onPress}
			onPressIn={animateInMarginBottom}
			onPressOut={animateOutMarginBottom}
		>
			<View
				style={[
					styles.button,
					passedClass,
					disabled && styles.buttonDisable,
				]}
			>
				<Text
					style={[
						styles.text,
						disabled && styles.buttonDisableText,
					]}
				>
					{title}
				</Text>
			</View>
			<Animated.View
				style={[
					!disabled && styles.buttonBackgroundAnimated,
					{
						marginLeft: marginAnimated,
						marginTop: marginAnimated,
					},
				]}
			/>
		</Pressable>
	)
}
