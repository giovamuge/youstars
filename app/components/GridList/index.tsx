import React from "react"
import { FlatList, View } from "react-native"
import { styles } from "./style"

type Props = {
	data: ArrayLike<any> | null | undefined
	columns: number
	ListEmptyComponent?: () => JSX.Element
	ListFooterComponent?: () => JSX.Element
	onEndReachedThreshold?: number
	onEndReached?: () => void
	renderItem: ({
		item,
		index,
	}: {
		item: any
		index: number
	}) => JSX.Element
}

/**
 * Render a list of items in a grid settings column with FlatList
 */
export function GridList({
	data,
	renderItem,
	columns = 1,
	ListEmptyComponent,
	ListFooterComponent,
	onEndReachedThreshold,
	onEndReached,
}: Props): JSX.Element {
	let myDataWrapped: any = []

	if (data && Array.isArray(data)) {
		for (let j = 0; j < data.length; j += columns) {
			const row = data.slice(j, j + columns)
			myDataWrapped.push(row)
		}
	}

	const customRenderItem = (item: any, index: number) => {
		const flex = item.length > 0 ? item.length : 1
		return (
			<View key={index} style={[styles.container, { flex }]}>
				{item &&
					item.length > 0 &&
					item.map((subItem: any, subIndex: number) => (
						<View
							style={styles.itemContainer}
							key={subIndex}
						>
							{renderItem({
								item: subItem,
								index: subIndex,
							})}
						</View>
					))}
			</View>
		)
	}

	return (
		<FlatList
			data={myDataWrapped}
			renderItem={listWrapped =>
				customRenderItem(listWrapped.item, listWrapped.index)
			}
			ListEmptyComponent={ListEmptyComponent}
			ListFooterComponent={ListFooterComponent}
			onEndReachedThreshold={onEndReachedThreshold}
			onEndReached={onEndReached}
		/>
	)
}

export default GridList
