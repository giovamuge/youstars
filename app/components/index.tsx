export * from "./Section/index"
export * from "./TextField/index"
export * from "./GridList/index"
export * from "./Button/index"
export * from "./Avatar/index"
export * from "./Header/index"
export * from "./BackButton/index"
