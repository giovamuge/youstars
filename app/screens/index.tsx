export * from "./Home/index"
export * from "./Stargazers/index"
export * from "./Profile/index"
