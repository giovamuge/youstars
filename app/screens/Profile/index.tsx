import React, { useEffect, useState } from "react"
import { SafeAreaView, ScrollView, Text, View } from "react-native"
import { User } from "../../models/user"
import { get } from "../../api/github/profile/get/index"
import { Avatar, Section } from "../../components"
import { typographyStyle } from "../../theme/typography"
import { Header } from "../../components/Header"
import { styles } from "./style"

function Profile({
	// navigation,
	route,
}: {
	navigation: any
	route: any
}): JSX.Element {
	const { owner } = route.params
	const [profile, setProfile] = useState<User>()

	const getProfile = async () => {
		try {
			const user = await get(owner)
			console.log("profile: ", user)
			setProfile(() => user)
		} catch (error) {
			console.log(error)
		}
	}

	useEffect(() => {
		getProfile()
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [])

	const renderSection = (value: string, title: string) => {
		return (
			value && (
				<Section title={title}>
					<Text style={typographyStyle.text3}>{value}</Text>
				</Section>
			)
		)
	}

	return (
		<SafeAreaView>
			<Header backTitle="home" />
			<ScrollView>
				{profile ? (
					<View style={styles.container}>
						<View style={styles.avatarContainer}>
							{profile.avatar_url && (
								<Avatar
									size={100}
									uri={profile.avatar_url ?? ""}
								/>
							)}
						</View>

						<Text
							style={[
								typographyStyle.title,
								styles.title,
							]}
						>
							{profile.name}
						</Text>

						<Text style={typographyStyle.text2Bold}>
							{profile.login}
						</Text>

						<View style={styles.additionalContainer}>
							{renderSection(profile.bio, "Bio")}
							{renderSection(profile.email, "Email")}
							{renderSection(profile.blog, "Blog")}
							{renderSection(
								profile.location,
								"Location",
							)}
							{renderSection(
								profile.company,
								"Company",
							)}
							{renderSection(
								profile.twitter_username,
								"Twitter",
							)}
							{renderSection(
								profile.followers.toString(),
								"Followers",
							)}
							{renderSection(
								profile.following.toString(),
								"Following",
							)}

							{
								// TODD: implement repos list in profile
							}
						</View>
					</View>
				) : (
					<Text style={typographyStyle.title}>
						Loading...
					</Text>
				)}
			</ScrollView>
		</SafeAreaView>
	)
}

export default Profile
