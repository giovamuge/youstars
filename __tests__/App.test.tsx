/**
 * @format
 */

import "react-native"
import React from "react"

// Note: import explicitly to use the types shipped with jest.
import { it } from "@jest/globals"

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer"
import { Home, Stargazers } from "../app/screens"
import { GridList } from "../app/components"
import { Text } from "react-native"
import Profile from "../app/screens/Profile"

const navigation = {
	//@ts-ignore
	navigate: jest.fn(),
}

it("renders Home page", () => {
	const props = { navigation }
	const tree = renderer.create(<Home {...props} />).toJSON()
	//@ts-ignore
	expect(tree).toMatchSnapshot()
})

it("renders StarGazers page", () => {
	const props = {
		route: {
			params: { owner: "giovamuge", repo: "stadio-visione-3d" },
		},
		navigation,
	}
	const tree = renderer.create(<Stargazers {...props} />).toJSON()
	//@ts-ignore
	expect(tree).toMatchSnapshot()
})

it("renders GridList component", () => {
	const data = [{ name: "test 1" }, { name: "test 2" }]
	const tree = renderer
		.create(
			<GridList
				columns={2}
				data={data}
				renderItem={({ item }) => {
					return <Text>{item.name}</Text>
				}}
			/>,
		)
		.toJSON()
	//@ts-ignore
	expect(tree).toMatchSnapshot()
})

it("renders Profile page", () => {
	const props = {
		route: {
			params: { owner: "giovamuge" },
		},
		navigation,
	}
	const tree = renderer.create(<Profile {...props} />).toJSON()
	//@ts-ignore
	expect(tree).toMatchSnapshot()
})
