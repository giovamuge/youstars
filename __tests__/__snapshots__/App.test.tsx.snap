// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GridList component 1`] = `
<RCTScrollView
  data={
    [
      [
        {
          "name": "test 1",
        },
        {
          "name": "test 2",
        },
      ],
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          [
            {
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
            },
            {
              "flex": 2,
            },
          ]
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text>
            test 1
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text>
            test 2
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders Home page 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <RCTScrollView
    style={
      {
        "height": "100%",
        "paddingHorizontal": 20,
        "paddingTop": 20,
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          [
            {
              "flexWrap": "wrap",
              "fontSize": 50,
              "fontWeight": "bold",
              "lineHeight": 45,
              "shadowColor": "#CCFF66",
              "shadowOffset": {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
            },
            {
              "marginBottom": 15,
            },
          ]
        }
      >
        Wich stars would you know?
      </Text>
      <View
        style={
          {
            "flexDirection": "column",
            "gap": 7,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 24,
              "shadowColor": "#CCFF66",
              "shadowOffset": {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
            }
          }
        >
          Owner
        </Text>
        <View
          style={
            {
              "marginBottom": 20,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#CCFF66",
                "justifyContent": "center",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#000000",
                  "borderWidth": 3,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "padding": 20,
                  "width": "100%",
                }
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="meta"
                placeholderTextColor="#A8A8A8"
                secureTextEntry={false}
                selectionColor="#CCFF66"
                style={
                  {
                    "color": "#000000",
                    "height": 50,
                    "width": "100%",
                  }
                }
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "column",
            "gap": 7,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 24,
              "shadowColor": "#CCFF66",
              "shadowOffset": {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
            }
          }
        >
          Repository
        </Text>
        <View
          style={
            {
              "marginBottom": 20,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#CCFF66",
                "justifyContent": "center",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#000000",
                  "borderWidth": 3,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "padding": 20,
                  "width": "100%",
                }
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="react-native"
                placeholderTextColor="#A8A8A8"
                secureTextEntry={false}
                selectionColor="#CCFF66"
                style={
                  {
                    "color": "#000000",
                    "height": 50,
                    "width": "100%",
                  }
                }
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "fontSize": 13,
              "fontWeight": "500",
              "lineHeight": 15.6,
              "shadowColor": "#CCFF66",
              "shadowOffset": {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
            },
            {
              "marginBottom": 15,
              "textAlign": "center",
              "width": "100%",
            },
          ]
        }
      >
        alternatively
      </Text>
      <View
        style={
          {
            "flexDirection": "column",
            "gap": 7,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 24,
              "shadowColor": "#CCFF66",
              "shadowOffset": {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
            }
          }
        >
          Url repository
        </Text>
        <View
          style={
            {
              "marginBottom": 20,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#CCFF66",
                "justifyContent": "center",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#000000",
                  "borderWidth": 3,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "padding": 20,
                  "width": "100%",
                }
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="http://github.com/meta/react-native"
                placeholderTextColor="#A8A8A8"
                secureTextEntry={false}
                selectionColor="#CCFF66"
                style={
                  {
                    "color": "#000000",
                    "height": 50,
                    "width": "100%",
                  }
                }
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignSelf": "flex-start",
                "borderColor": "#000000",
                "borderWidth": 3,
                "width": "100%",
              },
              undefined,
              {
                "backgroundColor": "#EBEBEB",
                "borderColor": "#A8A8A8",
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginHorizontal": 48,
                  "marginVertical": 12,
                  "textAlign": "center",
                },
                {
                  "color": "#A8A8A8",
                },
              ]
            }
          >
            Search
          </Text>
        </View>
        <View
          collapsable={false}
          style={
            {
              "marginLeft": 7,
              "marginTop": 7,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`renders StarGazers page 1`] = `
<View
  style={
    {
      "backgroundColor": "#CCFF66",
      "flex": 1,
      "height": "100%",
      "paddingVertical": 50,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    ListFooterComponent={[Function]}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.2}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 13,
              "fontWeight": "700",
              "lineHeight": 15.6,
              "shadowColor": "#CCFF66",
              "shadowOffset": {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
            }
          }
        >
          There are no stars in the set parameters
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 10,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
